src/js/controller/setup-steps.js

```diff
@@ -1,231 +1,123 @@
+                error: err
+            });
+        });
+        pluginLoader.load();
+    });
+}
+
+function initPlugins(_model, _api, _view) {
+    return Promise.all([
+        setupView(_model, _view),
+        loadPlugins(_model)
+    ]).then(() => {
+        // TODO: check destroyed
         delete window.jwplayerPluginJsonp;
-        _pluginLoader.setupPlugins(_api, _model);
-        resolve();
-    }
-
-    function _pluginsError(resolve, evt) {
-        error(resolve, 'Could not load plugin', evt.message);
-    }
-
-    function _loadPlaylist(resolve, _model) {
-        var playlist = _model.get('playlist');
-        if (_.isString(playlist)) {
-            _playlistLoader = new PlaylistLoader();
-            _playlistLoader.on(PLAYLIST_LOADED, function(data) {
-                _model.attributes.feedData = data;
-                _model.attributes.playlist = data.playlist;
-                resolve();
+        pluginLoader.setupPlugins(_api, _model);
+    });
+}
+
+function loadPlaylist(_model) {
+    const playlist = _model.get('playlist');
+    if (_.isString(playlist)) {
+        return new Promise((resolve, reject) => {
+            const playlistLoader = new PlaylistLoader();
+            playlistLoader.on(PLAYLIST_LOADED, function(data) {
+                resolve(data.playlist, data);
             });
-            _playlistLoader.on(ERROR, _.partial(_playlistError, resolve));
-            _playlistLoader.load(playlist);
-        } else {
-            resolve();
-        }
-    }
-
-    function _filterPlaylist(resolve, _model, _api, _view, _setPlaylist) {
-        // Performs filtering
-        var success = _setPlaylist(_model.get('playlist'), _model.get('feedData'));
+            playlistLoader.on(ERROR, err => {
+                reject(new Error(`Error loading playlist: ${err.message}`));
+            });
+            playlistLoader.load(playlist);
+        });
 
-        if (success) {
-            resolve();
-        } else {
-            _playlistError(resolve);
-        }
     }
+    return Promise.resolve(playlist, _model.attributes.feedData);
+}
 
-    function _playlistError(resolve, evt) {
-        if (evt && evt.message) {
-            error(resolve, 'Error loading playlist', evt.message);
-        } else {
-            error(resolve, 'Error loading player', 'No playable sources found');
-        }
-    }
+function filterPlaylist(_model) {
+    return loadPlaylist(_model).then((playlist, feedData) => {
+        // TODO: check destroyed
+        // `setPlaylist` performs filtering
+        setPlaylist(_model, playlist, feedData);
+        loadProvidersForPlaylist(_model);
+    });
+}
 
-    function isSkinLoaded(skinPath) {
-        var ss = document.styleSheets;
-        for (var i = 0, max = ss.length; i < max; i++) {
-            if (ss[i].href === skinPath) {
-                return true;
-            }
+function isSkinLoaded(skinPath) {
+    const ss = document.styleSheets;
```

Comment:
I ended up not removing it, since we need them for when custom skins passed in the config